mindmeld/components/dialogue.py

```diff
@@ -689,6 +692,249 @@ def _get_dialogue_handler(self, dialogue_state):
         return handler
 
 
+class AutoEntityFilling:
+    """A special dialogue flow sublcass to implement Automatic Entitiy (Slot) Filling
+    (AEF) that allows developers to prompt users for completing the missing
+    requirements for entity slots.
+
+    Attributes:
+        app (Application): The application that initializes this flow.
+    """
+
+    _logger = mod_logger.getChild("AutoEntityFilling")
+    """Class logger."""
+
+    def __init__(self, entrance_handler, form, app):
+        self._app = app
+        self._entrance_handler = entrance_handler
+        self._form = form
+        self._local_form = None
+        self._prompt_user = None
+        self._check_attr()
+
+    def _check_attr(self):
+        if not ('entities' in self._form and len(self._form['entities']) > 0):
+            raise KeyError("Entity list cannot be empty.")
+
+        self._entity_form = self._form['entities']
+        self._max_retries = (
+            self._form['max_retries'] if 'max_retries' in self._form else 1)
+        self._exit_response = (
+            self._form['exit_msg'] if 'exit_msg' in self._form else 'How may I help you?')
+        self._exit_intent = (
+            self._form['exit_intent'] if 'exit_intent' in self._form else None)
+        self._exit_keys = (
+            self._form['exit_keys'] if 'exit_keys' in self._form else
+            ['cancel', 'restart', 'exit', 'reset'])
+
+        if not isinstance(self._max_retries, int):
+                    raise TypeError("'max_retries' should be of type: int.")
```

Comment:
(minor) indentation looks off here - did the changes pass lint checks?

Reply:
Strange, it did pass the lint checks.

Updated it anyhow!